### ISSUE TITLE
AttributeError: 'module' object has no attribute 'compress'

### ISSUE THREAD

**Issue by danroestorf** [DataDog/datadog-serverless-functions#201]
**Describe what happened:**

A fresh setup of serverlessrepo-Datadog-Log-Forwarder now results in an error message when using it with a CloudWatch trigger.

```
[ERROR]	2020-02-05T19:04:31.104Z Exception while forwarding logs in batch 
...
Traceback (most recent call last):
  File "/var/task/lambda_function.py", line 544, in forward_logs
    client.send(batch)
  File "/var/task/lambda_function.py", line 295, in send
    self._client.send(logs)
  File "/var/task/lambda_function.py", line 395, in send
    data = compress_logs(data, DD_COMPRESSION_LEVEL)
  File "/var/task/lambda_function.py", line 475, in compress_logs
    return gzip.compress(bytes(batch, 'utf-8'), compression_level)
AttributeError: 'module' object has no attribute 'compress'
```

Searching for a solution for this came up with no results. Upon reading https://github.com/DataDog/datadog-serverless-functions/pull/190 and sifting through the Lambda code it turns out that setting `DD_USE_COMPRESSION=false` is now required for an out of the box installation to work.

**Describe what you expected:**

A fresh setup of the log forwarder Lambda should work with a simple configuration.

**Steps to reproduce the issue:**

Spin up a new Lambda and setup a CloudWatch log group trigger.

**Comment by tianchu** [comment]
Hi @DanRoestorf, thanks for the issue report! Which version of the forwarder you are using (you can find it in `lambda_function.py`)?

**Comment by danroestorf** [comment]
@tianchu `DD_FORWARDER_VERSION = "2.4.0"`

**Comment by tianchu** [comment]
I'm not sure why would `gzip` not have `compress` method..., but FYI, forwarder 3.x is now available, it's provisioned by CloudFormation instead of SAM. Do you mind giving it shot and see if same issue persists? Here are the steps for upgrade to 3.x: https://github.com/DataDog/datadog-serverless-functions/blob/ff361aaeea0d01602e48666ee8ddbeac8aae3b31/aws/logs_monitoring/README.md#upgrade-an-older-version-to-300

**Comment by tianchu** [comment]
@DanRoestorf does upgrading the forwarder fix the issue? If not, I suspect it's a python runtime issue, you can try updating your forwarder Lambda's runtime to Python3.7.

**Comment by Y4suyuki** [comment]
Hi @tianchu. I had the same issue. Are you sure version 2.4.0 work fine?
compress attribute of gzip is new in python 3.2 according to python docs and template
set runtime to python2.7. So it looks like version 2.4.0 is not working unless we set `DD_USE_COMPRESSION=false`.
Or sam version of datadog log forwarder is deprecated and not maintained?

**Comment by tianchu** [comment]
Yes, I can confirm that version 2.4.0 does not work for python2.7 and the SAM application is no longer maintained.

Sorry for the confusion, I believe v2.4.0 was released amid the development of v3.0.0, and it was not well tested against python2.7, which is deprecated. I will release a v2.4.1 patch to the SAM application and set the runtime to python3.7.

We have upgraded to python3.7 since v3.0.0, which offers improved installation and upgrade experience through CloudFormation, please follow these instructions to upgrade: https://github.com/DataDog/datadog-serverless-functions/tree/master/aws/logs_monitoring#upgrade-an-older-version-to-300
